src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java

```diff
@@ -951,6 +962,132 @@ private static void fetchFromValueSource(
+          } else if (obj instanceof Boolean) {
+            compositeFieldValue.addFieldValue(
+                SearchResponse.Hit.FieldValue.newBuilder().setBooleanValue((Boolean) obj));
+          } else if (obj instanceof ArrayList) {
+            SearchResponse.Hit.FieldValue.Builder fieldValueBuilder =
+                SearchResponse.Hit.FieldValue.newBuilder();
+            SearchResponse.Hit.FieldValue.RepeatedFieldValues.Builder repeatedFieldValuesBuilder =
+                SearchResponse.Hit.FieldValue.RepeatedFieldValues.newBuilder();
+            generateRepeatedFields(repeatedFieldValuesBuilder, obj);
+            fieldValueBuilder.setRepeatedFieldValues(repeatedFieldValuesBuilder);
+            compositeFieldValue.addFieldValue(fieldValueBuilder);
+          }
+          hit.putFields(name, compositeFieldValue.build());
+        }
+      }
+    }
+
+    /**
+     * Generates list fields for all types except for lists.
+     *
+     * <p>TODO: Support list of lists.
+     *
+     * @param repeatedFieldValuesBuilder The builder for repeated field value in the search
+     *     response.
+     * @param objs Runtime field objects that are supposed to be list.
+     */
+    private static void generateRepeatedFields(
+        SearchResponse.Hit.FieldValue.RepeatedFieldValues.Builder repeatedFieldValuesBuilder,
+        Object objs) {
+      List<Struct> structObjs = new ArrayList<>();
+      for (Object obj : (Iterable<? extends Object>) objs) {
+        if (obj instanceof String) {
+          repeatedFieldValuesBuilder.addAllTextValues((Iterable<String>) objs);
+          return;
+        } else if (obj instanceof Integer) {
+          repeatedFieldValuesBuilder.addAllIntValues((Iterable<Integer>) objs);
+          return;
+        } else if (obj instanceof Double) {
+          repeatedFieldValuesBuilder.addAllDoubleValues((Iterable<Double>) objs);
+          return;
+        } else if (obj instanceof Float) {
+          repeatedFieldValuesBuilder.addAllFloatValues((Iterable<Float>) objs);
+          return;
+        } else if (obj instanceof Long) {
+          repeatedFieldValuesBuilder.addAllLongValues((Iterable<Long>) objs);
+          return;
+        } else if (obj instanceof Boolean) {
+          repeatedFieldValuesBuilder.addAllBooleanValues((Iterable<Boolean>) objs);
+          return;
+        } else if (obj instanceof Map) {
+          structObjs.add(createStruct((Map<String, Object>) obj));
+        } else {
+          return;
+        }
+      }
+
+      // Construct repeated structs.
+      if (Iterables.size(structObjs) > 0) {
+        repeatedFieldValuesBuilder.addAllStructValues(structObjs);
+      }
+    }
+
+    /**
+     * Create a struct for the given map based on object value type
+     *
+     * @param map The given map
+     * @return Struct The protobuf representation of the map using a struct object.
+     */
+    private static Struct createStruct(Map<String, Object> map) {
```

Comment:
You should be able to use https://github.com/Yelp/nrtsearch/blob/master/clientlib/src/main/java/com/yelp/nrtsearch/server/utils/StructJsonUtils.java#L46 to convert the map